### PR TITLE
add weight getter and setter to dmp base class

### PR DESCRIPTION
**Pull request by maotto**

### CODE REVIEW

File: movement_primitives/dmp/_base.py [hunk 1]
```diff
@@ -15,3 +15,24 @@ def reset(self):
         self.last_t = None
         self.current_y = np.copy(self.start_y)
         self.current_yd = np.copy(self.start_yd)
+
+    def get_weights(self):
```
Comment (AlexanderFabisch) [inline movement_primitives/dmp/_base.py:19]: * I would still prefer using another mixin for this.
* Also the members forcing_term and n_weights_per_dim are not defined.
* A docstring should not contain information about subclasses.

File: movement_primitives/dmp/_base.py [hunk 2]
```diff
@@ -16,23 +16,33 @@ def reset(self):
...
+class WeightParametersMixin:
+    """Mixin class providing common access methods to forcing term weights."""
     def get_weights(self):
         """Get weight vector of DMP.
 
         Returns
         -------
         weights : array, shape (N * n_weights_per_dim,)
-            Current weights of the DMP. N depend on the type of DMP (task space dim. or for dual-cartesian=12)
+            Current weights of the DMP. N depends on the type of DMP
```
Comment (AlexanderFabisch) [inline movement_primitives/dmp/_base.py:28]: This does not necessarily have to be a DMP, could also be a ProMP!

File: movement_primitives/dmp/_dmp.py [hunk 3]
```diff
@@ -296,7 +296,7 @@ def dmp_step_euler(
     pass
 
 
-class DMP(DMPBase):
+class DMP(WeightParametersMixin, DMPBase):
```
Comment (AlexanderFabisch) [inline movement_primitives/dmp/_dmp.py:299]: Is there a reason for this order?

Comment (maotto) [reply movement_primitives/dmp/_dmp.py:299]: inspired by [scikit](https://github.com/scikit-learn/scikit-learn/blob/bbd64c16f403f6b1b982c95673324db84b1196e1/sklearn/gaussian_process/_gpr.py#L24)

File: movement_primitives/dmp/_base.py (hunk 2 above)
Comment (AlexanderFabisch) [reply movement_primitives/dmp/_base.py:28]: On the other hand, the default implementation uses self.forcing_term, which is very specific for DMPs... I don't know what to do here.

Comment (maotto) [reply movement_primitives/dmp/_base.py:28]: more general terms could be movement_primitive, movement class, representation or behavior.

File: movement_primitives/dmp/_dmp.py (hunk 3 above)
Comment (maotto) [reply movement_primitives/dmp/_dmp.py:299]: or as [they](https://realpython.com/lessons/multiple-inheritance-python/) say, "A mixin is a class that doesn’t care about its position in the hierarchy, but just provides one or more convenience methods"

Comment (AlexanderFabisch) [reply movement_primitives/dmp/_dmp.py:299]: OK, thanks

File: movement_primitives/dmp/_base.py (hunk 2 above)
Comment (AlexanderFabisch) [reply movement_primitives/dmp/_base.py:28]: yep, but in that case we would have to move this to movement_primitive.base and then we have DMP-specific code there...

**Pull request merged**